feat(button): add pressed/selected state styling

kmundlamuri opened the pull request:
**This change:** (check at least one)

- [x] Adds a new feature
- [ ] Fixes a bug
- [ ] Improves maintainability
- [ ] Improves documentation
- [ ] Is a release activity

**Is this a breaking change?** (check one)

- [ ] Yes
- [ ] No

**Is the:** (complete all)

- [ ] Title of this pull request clear, concise, and indicative of the issue number it addresses, if any?
- [ ] Test suite(s) passing?
- [ ] Code coverage maximal?
- [ ] Changeset added?
- [ ] Component status page up to date?

**What does this change address?**
Resolves #995 

**How does this change work?**
A clear, detailed description of how this change addresses the issue.
This could be a bullet list if there are several moving parts.

**Additional context**
Add any other context here.

daneah commented on packages/pharos/package.json:
```diff
@@ -58,6 +58,7 @@
   "dependencies": {
     "@floating-ui/dom": "^1.7.2",
     "@ithaka/focus-trap": "2.0.1",
+    "@ithaka/pharos": "file:.yalc/@ithaka/pharos",
```
```suggestion
```

daneah commented on packages/pharos/src/components/toggle-button-group/pharos-toggle-button.ts:
```diff
@@ -20,7 +20,7 @@ export class PharosToggleButton extends PharosButton {
    */
   @property({ type: Boolean, reflect: true })
-  public selected = false;
+  public override selected = false;
```
Perhaps this no longer needs to be present, if inheriting it from the superclass now?

brentswisher commented:
Preview available here, we can trigger this manually now but doesn't seem to add a status check, which makes sense: https://60919c26122bd50039b34644-uczdkxjbjc.chromatic.com/?path=/story/components-button--selected

kmundlamuri commented:
> Hey @kmundlamuri - sorry again for the delay, this sparked quite the discussion with the maintainers deciding the best way forward 😄
> 
> Here is what we are thinking:
> 
> * Instead of adding a new attribute for `selected` we would like to use the existing `a11yPressed` attribute, and add the styles based on this. (Since we would never want `selected` and `a11yPressed` to be out of synch, it makes more sense to lean into what we already have.)
> * This will be a little more complicated as the `toggle-button-group` has a `toggle-button` which extends from the base button, but needs to maintain separate styles.
> 
> So to move forward, we want to rework you PR to add styles to the existing `a11yPressed` attribute, and then we will likely also need to adjust the CSS of the toggle button to override those so that it remains styled as it currently is.
> 
> I know that is a little more work than you starting with, but I'm happy to pair with you Friday afternoon if it starts to get tricky, just let me know!

Thanks @brentswisher for the update and context! I’ve updated the PR to use the existing a11yPressed attribute for styling as discussed, and adjusted the toggle button CSS so that its current styling is preserved.

Please let me know if you’d like me to walk through the changes or if any further tweaks are needed.

sirrah-tam reviewed (approved):
Validated that the selected buttons convey the selected state programmatically (ie. passing down the `aria-pressed="true"` to the button) and that the visual styles of the toggle buttons are maintained while the new styles for selected buttons are added.

brentswisher commented on packages/pharos/src/components/button/pharos-button.wc.stories.ts:
```diff
@@ -289,3 +289,39 @@ export const Forms: Story = {
   `,
 };
+
+export const Selected: Story = {
```
I'm not sure have a new story for "selected" makes sense here any more, maybe we could update the variants to have a toggle-able state that sets/unsets a11y-pressed>

kmundlamuri commented:
> Left a few comments on things

Thanks @brentswisher I have made recommended changes, please let me know you thoughts

brentswisher commented:
@joiechang @sirrah-tam 
I kind of anticipated when I press on a button, it would show the pressed styles for as long as I am holding my mouse on it, but maybe I am misunderstanding the intent? I know we also want people to make it so the press "sticks" by toggling the a11y-pressed attribute, but should it always be styled as pressed too?
